internal/common/validation/podspec.go

```diff
@@ -54,6 +59,35 @@ func ValidatePodSpec(spec *v1.PodSpec, schedulingSpec *configuration.SchedulingC
 	return validatePorts(spec)
 }
 
+func validateTerminationGracePeriod(spec *v1.PodSpec, config *configuration.SchedulingConfig) error {
+	hasTerminationGracePeriod := spec.TerminationGracePeriodSeconds != nil
+	var terminationGracePeriodSeconds int64
+	var exceedsBounds, isDefault bool
+	if hasTerminationGracePeriod {
+		terminationGracePeriodSeconds = *spec.TerminationGracePeriodSeconds
+		exceedsBounds = (terminationGracePeriodSeconds < int64(config.TerminationGracePeriod.Minimum.Seconds()) ||
+			terminationGracePeriodSeconds > int64(config.TerminationGracePeriod.Maximum.Seconds()))
+		isDefault = terminationGracePeriodSeconds == int64(config.TerminationGracePeriod.Default)
```

Comment:
So this isn't necessarily true?  Just becuase the `terminationGracePeriodSeconds ` is the same as our default, it doesn't mean we defaulted it.  In the case that the user happened to explicitly pick a value that matches our defualt, we'll get a misleading error.

Regardless, I've made a comment below, saying that we shouldn't be sanity checking the default value in this function.

Reply:
removing

Reply:
removing here -- but in general the "applyDefaults" logic precedes validation, so it's not just this case where explicit vs default cannot be distinguished